src/sources/soundsourcem4a.cpp

```diff
@@ -506,6 +519,58 @@ void SoundSourceM4A::restartDecoding(MP4SampleId sampleBlockId) {
     m_sampleBuffer.clear();
 }
 
+int SoundSourceM4A::seekAndReadSampleBlock(
+        int sampleBlockId,
+        uint8_t** ppBytes,
+        uint32_t* pNumBytes) {
+    DEBUG_ASSERT(sampleBlockId > 0 && sampleBlockId <= m_pAvStream->nb_index_entries);
+
+    int error = 0;
+    char error_string[AV_ERROR_MAX_STRING_SIZE];
+
+    if (m_nextSampleBlockId != sampleBlockId) {
+        error = av_seek_frame(
+                m_pAvInputFormatContext,
+                m_pAvStream->index,
+                m_pAvStream->index_entries[sampleBlockId - 1].timestamp,
+                SEEK_SET);
+
+        if (error < 0) {
+            av_make_error_string(error_string, AV_ERROR_MAX_STRING_SIZE, error);
+            qWarning() << "Could not seek frame:" << error_string;
+            return error;
+        }
+    }
+
+    // create an empty package
+    AVPacket packet;
+    av_init_packet(&packet);
+    packet.data = nullptr;
+    packet.size = 0;
+
+    m_nextSampleBlockId = kSampleBlockIdInvalid;
+
+    error = av_read_frame(m_pAvInputFormatContext, &packet);
+    if (error < 0) {
+        av_make_error_string(error_string, AV_ERROR_MAX_STRING_SIZE, error);
+        qWarning() << "Could not read frame:" << error_string;
+    } else if (packet.size <= static_cast<int>(*pNumBytes)) {
+        memcpy(*ppBytes, packet.data, packet.size);
+        m_nextSampleBlockId = sampleBlockId + 1;
```

Comment:
Do you really need m_nextSampleBlockId for keeping track? I guess m_curSampleBlockId could be sufficient.